nodes/viz/viewer_draw_mk4.py

```diff
@@ -185,19 +185,20 @@ def view_3d_geom(context, args):
         bgl.glLineWidth(1)
 
     if config.draw_verts:
-        bgl.glPointSize(config.point_size)
-        if config.uniform_verts:
-            v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices})
-            config.v_shader.bind()
-            config.v_shader.uniform_float("color", config.vector_color[0][0])
-        else:
-            v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices, "color": geom.points_color})
-            config.v_shader.bind()
+        if len(geom.v_vertices)>0 and (len(geom.v_vertices[0])==3):
+            bgl.glPointSize(config.point_size)
+            if config.uniform_verts:
+                v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices})
+                config.v_shader.bind()
+                config.v_shader.uniform_float("color", config.vector_color[0][0])
+            else:
+                v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices, "color": geom.points_color})
+                config.v_shader.bind()
 
-        v_batch.draw(config.v_shader)
-        bgl.glPointSize(1)
+            v_batch.draw(config.v_shader)
+            bgl.glPointSize(1)
 
-    bgl.glEnable(bgl.GL_BLEND)
+    bgl.glDisable( bgl.GL_BLEND )
 
```

Comment:
It's not important but just to let you know that it also contradicts with pep8 :p

![image](https://github.com/nortikin/sverchok/assets/28003269/ba8b5ee5-24b4-4edf-b432-031f4c917e9d)

Reply:
I search code for “ .disable”. I’ll fix it.

Reply:
done